devtools/gradle/src/functionalTest/java/io/quarkus/gradle/FastJarFormatWorksTest.java

```diff
@@ -0,0 +1,155 @@
+        Assertions.assertTrue(Files.exists(jar));
+
+        File output = new File(projectDir, "target/output.log");
+        output.createNewFile();
+
+        Process process = doLaunch(jar, output);
+        try {
+            // Wait until server up
+            await()
+                    .pollDelay(1, TimeUnit.SECONDS)
+                    .atMost(1, TimeUnit.MINUTES).until(() -> DevModeTestUtils.getHttpResponse("/app/hello/package", 200));
+
+            String logs = FileUtils.readFileToString(output, "UTF-8");
+
+            assertThatOutputWorksCorrectly(logs);
+
+            // test that the application name and version are properly set
+            assertApplicationPropertiesSetCorrectly();
+            assertResourceReadingFromClassPathWorksCorrectly("");
+        } finally {
+            process.destroy();
+        }
+
+    }
+
+    private Process doLaunch(Path jar, File output) throws IOException {
+        List<String> commands = new ArrayList<>();
+        commands.add(JavaBinFinder.findBin());
+        commands.add("-jar");
+        commands.add(jar.toString());
+        ProcessBuilder processBuilder = new ProcessBuilder(commands.toArray(new String[0]));
+        processBuilder.redirectOutput(output);
+        processBuilder.redirectError(output);
+        return processBuilder.start();
+    }
+
+    static void assertResourceReadingFromClassPathWorksCorrectly(String path) {
+        try {
+            URL url = new URL("http://localhost:8080" + path + "/app/classpathResources");
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            // the default Accept header used by HttpURLConnection is not compatible with RESTEasy negotiation as it uses q=.2
+            connection.setRequestProperty("Accept", "text/html, *; q=0.2, */*; q=0.2");
+            if (connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
+                failResourcesFromTheClasspath();
+            }
+            try (BufferedReader br = new BufferedReader(
+                    new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
+                String output = br.readLine();
+                assertThat(output).isEqualTo("success");
+            }
+        } catch (IOException e) {
+            failResourcesFromTheClasspath();
+        }
+    }
+
+    private static void failResourcesFromTheClasspath() {
+        fail("Failed to assert that the application properly reads resources from the classpath");
+    }
+
+    // we don't use REST Assured because its bundled Groovy version clashes with Maven Invoker's (which is also used in this module)
+    static void assertApplicationPropertiesSetCorrectly() {
+        assertApplicationPropertiesSetCorrectly("");
+    }
+
+    // we don't use REST Assured because its bundled Groovy version clashes with Maven Invoker's (which is also used in this module)
+    static void assertApplicationPropertiesSetCorrectly(String prefix) {
```

Comment:
Same here

Reply:
Darn, it seems like this comment is out of context:

I meant that I am pretty sure our Gradle tests have a much better way of doing this test. See method `getHttpResponse` from `QuarkusDevGradleTestBase` (which should probably be moved to `QuarkusGradleTestBase`)

Reply:
On it